_topics/assumeutxo.md

```diff
@@ -0,0 +1,46 @@
+---
+title: AssumeUTXO
+
+## Required.  At least one category to which this topic belongs.  See
+## schema for options
+categories:
+  - Consensus Enforcement
+  - P2P Network Protocol
+
+## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
+excerpt: >
+  **AssumeUTXO** is a proposed mode for bootstrapping new full
+  nodes without having them verify old block chain history.
```

Comment:
I think this should say that the node would verify old block chain history later (as it is, this summary suggests that assume UTXO nodes will never validate the historic blockchain). Perhaps: "mode for bootstrapping new full nodes that allows them to postpone verifying old block chain history." or similar.

Reply:
Hmm.  I forgot his proposal had that in it.  I've told him that I don't think that's generally useful.  E.g., see [my email to him](https://github.com/jamesob/assumeutxo-docs/blob/2019-04-proposal/proposal/feedback-david-harding.txt) with the quotes at the bottom of this reply.  I don't believe I've received a reply from him about the subject.

More specifically in reply to you, I'll mention the ability to background validate history later in the description.  I don't think it needs to go into the first sentence.

---

> I understand that some people want this, but I think it's not very
useful.  One of the really nice benefits of assumeutxo by itself is that
eliminates the wasted bandwidth and CPU of people verifying history that
they probably don't care about.  I think it'd be great to have an option
to verify old history (e.g. just like current -reindex-chainstate), but
it doesn't seem to me like it should be the default.

---

> I don't understand this argument.  If you just want to serve historic
blocks, you don't need to verify them.  You can put them in a torrent
(as has been done before).  Or you can simply download them, verify
they're part of the best headers chain, and share them without
verification.

---

> That new security model is implemented from the time you finish
assumeutxo+short-ibd until the time you finish genesis+full-ibd.  If
it's good enough for those hours/days/weeks/whatever, why isn't it good
enough all the time?

Reply:
> I'll mention the ability to background validate history later in the description. I don't think it needs to go into the first sentence.

The proposal as it stands is to always verify the historic chain in the background, and there are definitely some people who'd oppose the proposal if that wasn't the case. I think the current excerpt "a proposed mode for bootstrapping new full nodes without having them verify old block chain history." doesn't just omit that detail - it's actively misleading.

The proposal may eventually change to not default to background historic verification, but until that happens, we should summarize it as it exists now.

Reply:
> I think the current excerpt "a proposed mode for bootstrapping new full nodes without having them verify old block chain history." doesn't just omit that detail - it's actively misleading.

I'll change the text, but I'd like to say for posterity that I strongly disagree with your assessment that the current text is actively misleading.  (I'm also hurt by your characterization as such.)  Assumeutxo produces nodes that are functional from the users perspective---i.e. bootstrapped---but which have not verified old block chain history.  Just because the node may verify old chain history at some later point doesn't change the fact that the core proposal---the thing being summarized in this first sentence---is for people to be able to use a full node without having verified the full chain history.

Reply:
Oh no, I'm sorry Dave! That was a poor choice of words on my part. I don't think you're trying to mislead people. What I meant was that your original summary would lead people to (what I think is) the wrong conclusion.

I totally understand your argument that this *does* bootstrap nodes without verifying the full chain history. I also think it's important to note that this state (running a node without full history verification) is intended to be a temporary.

Reply:
John, thanks for the clarification and sorry that I took it so personally.  As mentioned, I'll update the summary as you requested.